draft-ietf-quic-transport.md

```diff
@@ -3998,6 +3998,46 @@ failed validation as a connection error of type TRANSPORT_PARAMETER_ERROR.
 A Retry packet does not include a packet number and cannot be explicitly
 acknowledged by a client.
 
+### 1-RTT Long Header Packets {#packet-1rtt-long}
+
+1-RTT packets can be used once the 1-RTT keys are negotiated.  Two types of
+1-RTT packet are defined: the 1-RTT long header packet and the short header
+packets {{short-header}}.
+
+Use of the short header packets is generally preferable, as it has less overhead
+and also exposes the Spin Bit.  The purpose of using the 1-RTT long header
+packets is to grease the Version field.
+
+~~~
++-+-+-+-+-+-+-+-+
+|1|1|R R R R|P P|
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         Version (32)                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+| DCID Len (8)  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|               Destination Connection ID (0..160)            ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+| SCID Len (8)  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                 Source Connection ID (0..160)               ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                           Length (i)                        ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                    Packet Number (8/16/24/32)               ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          Payload (*)                        ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+
+When using the 1-RTT long header format, the sender MUST set the Version field
+to a value other than 0x00000001.  The version SHOULD be picked at-random after
```

Comment:
We should allow 0x00000001 as the version number, because any version number should be usable for greasing, including the version numbers that have meanings.

And actually, this observation exposes an issue in the proposal.

Consider the case where a stateless load balancer receives a QUIC long header packet with the version set to 0x00000001. It could either be a 1-RTT packet with the version field being greased, or it could be an Initial packet of a new connection. The stateless load balancer cannot tell which is the case. In other words, trial decryption is required in certain circumstances.

Therefore, I think we would need to negotiate the use of long header 1-RTT packets. For example, we could use a TP for providing the peer the alternate version number that can be used for that particular connection.

Reply:
Addressed in 4f3c2ff.